src/movies.js

```diff
@@ -1,16 +1,46 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
+
+const movies = require("./data");
+
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+  const allDirectors = moviesArray.map(function(movies){
+      return movies.director
+  });
+  return allDirectors
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+  const spielbergDramas = moviesArray.filter(function(movies) {
+    return (movies.director === 'Steven Spielberg' && movies.genre.includes('Drama'))
+  })
+  return spielbergDramas.length
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+  const totalScore = moviesArray.reduce(function(accumulator, currentValue) {
+    return accumulator + currentValue.score
```

Comment:
some of movies don't have any score. because of that, you should check if movie has a score or not:

if(currentValue.score) {
      return  accumulator + currentValue.score
    }else {
      currentValue.score = 0
      return  accumulator + currentValue.score
    }